### PR TITLE
Rewrite Configable docstrings

### PR DESCRIPTION
**Pull request by vEpiphyte**
Update configable docstrings.  Biggest note is the addition of the race condition warning for onConfOptSet() functions.

### CODE REVIEW

File: synapse/lib/config.py [hunk 1]
```diff
@@ -112,36 +151,66 @@ def addConfDef(self, name, **info):
     def reqConfOk(self, opts):
         '''
         Check that that config values pass validation or raise.
+
+        Args:
+            opts (dict): Dictionary containing name, valu pairs to validate.
+
+        Returns:
```
Comment (vertexmc) [inline synapse/lib/config.py]: should this be `Raises:`?

File: synapse/lib/config.py [hunk 2]
```diff
@@ -178,14 +263,20 @@ def setConfOpts(self, opts):
...
+
+        Examples:
+            Set a pair of keys on a object using a dictionary::
 
-            opts = {
-                'foo:enabled':True,
-                'foo:size':1000
-            }
+                opts = {
+                    'foo:enabled':True,
```
Comment (vertexmc) [inline synapse/lib/config.py]: should there be a space between : and True? (not sure if style should be enforced in docstrings)

File: synapse/lib/config.py (hunk 1 above)
Comment (vEpiphyte) [reply synapse/lib/config.py]: yeah - good catch! Thanks!

File: synapse/lib/config.py (hunk 2 above)
Comment (vEpiphyte) [reply synapse/lib/config.py]: yeah - we should be good citizens there.

**Pull request merged**